doc/compiler/ProblemDetermination.md

```diff
@@ -229,286 +244,282 @@ when it ends. If the compiler fails on a particular method (that is, it starts
+binary encoding has been performed, annotated with the actual code bytes encoded for the
+instructions, and the absolute addresses of the instructions in the code cache. This
+listing also includes additional instructions like the prologue, the epilogue, and
+out-of-line snippet code. In other words, it includes all instructions that consistute 
+the jitted method. This listing should look fairly close to a disassembly listing that
+one can obtain from a debugger.
+**Method Meta-data:** A dump of meta-data about the method, such as GC atlas, exception
+ranges and handlers, etc.
+**Mixed Mode Disassembly:** The same listing as Post Binary Instructions, but annotated
+with Java bytecode call stack information. This makes it easy to see what native
+instructions were generated for a particular Java bytecode instruction.
+
+**Trees Listing**  
+
+The IL used in the Testarossa JIT compiler consists of a doubly linked list of trees of
+nodes. Nodes are identified by their memory addresses, and their functions are described 
+by their opcodes. Every node is either associated with a data type that indicates the 
+type of expression represented by the node, or marked as "NoType", i.e. evaluating the 
+node's opcode returns no value. Conceptually, each node can have multiple children, and 
+either zero or one parent node; parent-less nodes (or root nodes) become children of 
+"treetops", which are used to organize individual trees into a list. In reality, 
+commoned nodes have more than one parent, and every node has a reference count that is 
+used to keep track of how many parents point to itself.
+
+The following snippet contains part of the listing of generated trees for method
+`SimpleLooper.main([Ljava/lang/String;)V`:
+
+```
+Call Stack Info
+CalleeIndex CallerIndex ByteCodeIndex CalleeMethod
+       0         -1          b        SimpleLooper.<init>V
+       1          0          1        java/lang/Object.<init>V
+       2         -1         15        SimpleLooper.createArray(I)V
+       3         -1         19        SimpleLooper.walkArray()V
+       4         -1         2b        SimpleLooper.counter()I
+
+    ------------ ByteCodeIndex
+    |   ------------- CallSiteIndex
+    |   |   ------------- Reference Count
+    |   |   |     -------------- Visit Count
+    |   |   |     |     -------------- Global Index
+    |   |   |     |     |     ------------ Side Table Index
+    |   |   |     |     |     |   ------------- Use/def Index
+    |   |   |     |     |     |   |  ------------- Number of Children
+    |   |   |     |     |     |   |  |  ------------- Size
+    |   |   |     |     |     |   |  |  |      ------------- Data Type
+    |   |   |     |     |     |   |  |  |      |           ------------- Node Address
+    |   |   |     |     |     |   |  |  |      |           |      ------------- Instruction
+    |   |   |     |     |     |   |  |  |      |           |      |
+    V   V   V     V     V     V   V  V  V      V           V      V
```

Comment:
This section is outdated. The traces do not look like this anymore. Can we update to how they look today?